org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/XMLTextDocumentService.java

```diff
@@ -483,6 +519,19 @@ public void didSave(DidSaveTextDocumentParams params) {
 			// A document was saved, collect documents to revalidate
 			SaveContext context = new SaveContext(params.getTextDocument().getUri());
 			doSave(context);
+
+			// Manage didSave document lifecycle participants
+			final DOMDocument xmlDocument = getNowDOMDocument(params.getTextDocument().getUri());

```

Comment:
> This is done right after a save event, so its likely that the model is being recomputed. This may mean that the participant almost always receives a null xmlDocument. 

Sorry I don't understand. The DOM document is rebuild only didChange. When didSave is processed, it reuse the cached DOM document.

Reply:
Okay, I just misunderstood what was happening then. Please ignore my comment.